GOVERNANCE.md

```diff
@@ -0,0 +1,413 @@
+# Magic Modules Governance (a.k.a. "Golden Rules")
+
+This document specified the basic principles behind the Magic Modules and its
+"golden rules" that _shall not be infringed_ without a very strong (and
+documented) reason.
+
+In this doc the term _'must'_ means something that **have to** be followed at
+all times (aside some specific exceptions detailed in the
+[Exceptions](#exceptions) section), and _'should'_ means something that have to
+be followed **as much as possible**.
+
+
+## Table of Contents
+
+- [Folder Locations & Usage](#folder-locations--usage)
+- [Code Style](#code-style)
+- [Templates](#templates)
+- [Testing](#testing)
+- [Exceptions](#exceptions)
+- [Ruby Best Practices & Style Guide](#ruby-best-practices--style-guide)
+
+
+## Folder Locations & Usage
+
+  - `api`: Holds all definitions of objects used for defining products,
+    serialization and deserialization
+      * product independent
+      * provider independent
+  - `products/<product>/...`: Holds all definitions, files, helpers, etc
+    required to build the product, for all providers
+      * `api.yaml`
+          - provider independent
+      * `examples/<provider>`
+      * `helpers/`
+      * `files/`
+  - `provider`: All core provider features
+      * product independent
+      * provider independent
+  - `provider/<provider>/...`: Holds all provider specific code
+      * product independent
+  - `templates`: Holds all global templates
+      * product independent
+      * provider independent
+  - `templates/<provider>/...`: Holds all templates for the provider
+      * product independent
+
+In the list above folders without `...` means the folder only. `...` represents
```

Comment:
Move this and the 'Definitions' section directly under the 'folder/location' heading. The actual folder list should be presented after this intro material.

> In the list below, folders without the `...` means...